In [2]:
#!pip install iterative-stratification

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn import model_selection
from tqdm.auto import tqdm
from iterstrat.ml_stratifiers import MultilabelStratifiedKFold

ModuleNotFoundError: No module named 'iterstrat'

In [4]:
train = pd.read_csv('/Users/jasonrobinson/Documents/Projects/English-Language-Learning/data/train.csv')
test = pd.read_csv('/Users/jasonrobinson/Documents/Projects/English-Language-Learning/data/test.csv')
print(f"Train Shape:",train.shape)
print(f"Test Shape:",test.shape)

Train Shape: (3911, 8)
Test Shape: (3, 2)


In [ ]:
def get_new_labels(y):
y_new = LabelEncoder().fit_transform([''.join(str(l)) for l in y])
return y_new

labels_1 = get_new_labels(labels_train) skf = StratifiedKFold(n_splits=n_folds)
#mskf = MultilabelStratifiedKFold(n_splits=n_folds,shuffle=True,random_state=seed)
for train, test in skf.split(data_train,labels_1):
X_train = data_train[train]
X_test = data_train[test]
y_train = labels_train[train] # multilabel
y_test = labels_train[test]`

In [7]:
train.head()

text_id                                          full_text  cohesion  \
0  0016926B079C  I think that students would benefit from learn...       3.5   
1  0022683E9EA5  When a problem is a change you have to let it ...       2.5   
2  00299B378633  Dear, Principal\n\nIf u change the school poli...       3.0   
3  003885A45F42  The best time in life is when you become yours...       4.5   
4  0049B1DF5CCC  Small act of kindness can impact in other peop...       2.5   

   syntax  vocabulary  phraseology  grammar  conventions  
0     3.5         3.0          3.0      4.0          3.0  
1     2.5         3.0          2.0      2.0          2.5  
2     3.5         3.0          3.0      3.0          2.5  
3     4.5         4.5          4.5      4.0          5.0  
4     3.0         3.0          3.0      2.5          2.5

In [29]:
train['full_text'][4]

'Small act of kindness can impact in other people can change people to become better persons you can have an impact of kindess with a homeles that can change his life or with some who needed they are going to know you are a nice person if you are a nice person everywhere you go people is going to like your personality so you have to be a nice person with others like a old women triying to cross the road thats a impact of kindness when you do that you feel a greate person you can change people in the way they think by helping others treating nice other people give them some advice when you see someone need it help someone older then you give food and new clothes to a homeless person is a big act of kindess maybe you wount change his life but you would change one day of his life when you do thoose stuffs you feel emotional.\n\nAnother example of an act of kindess is when you help your friends to study for a test or explain them or when they need money you give some money is an act of kin

In [30]:
train.describe()

cohesion       syntax   vocabulary  phraseology      grammar  \
count  3911.000000  3911.000000  3911.000000  3911.000000  3911.000000   
mean      3.127077     3.028254     3.235745     3.116850     3.032856   
std       0.662542     0.644399     0.583148     0.655997     0.699841   
min       1.000000     1.000000     1.000000     1.000000     1.000000   
25%       2.500000     2.500000     3.000000     2.500000     2.500000   
50%       3.000000     3.000000     3.000000     3.000000     3.000000   
75%       3.500000     3.500000     3.500000     3.500000     3.500000   
max       5.000000     5.000000     5.000000     5.000000     5.000000   

       conventions  
count  3911.000000  
mean      3.081053  
std       0.671450  
min       1.000000  
25%       2.500000  
50%       3.000000  
75%       3.500000  
max       5.000000

The average score for all labels is 3, vocab is slightly averaging higher; however std is lower than others so likely some outliers in vocabulary.

## Tips-Advice

#### How To Submit TensorFlow Without Internet

Many people ask me, how do I submit TensorFlow models without internet? With HuggingFace Transformer, it's easy. Just download the following 3 things (1) model weights, (2) tokenizer files, (3) config file, and upload them to a Kaggle dataset. Below shows code how to get the files from HuggingFace for AllenAI's model longformer-base. But this same code can download any transformer, like for example roberta-base.

- TOKENIZER FILES - merges.txt, tokenizer_config.json, special_tokens_map.json, tokenizer.json, vocab.json
- CONFIG FILE - config.json
- MODEL WEIGHT FILE - tf_model.h5

Then just upload all these files to a Kaggle dataset, like what I did here. Then you load them into your notebook like the notebook you are reading. And we can turn internet off!

if DOWNLOADED_MODEL_PATH == 'model':
    os.mkdir('model')
    
    tokenizer = AutoTokenizer.from_pretrained(MODEL_NAME)
    tokenizer.save_pretrained('model')

    config = AutoConfig.from_pretrained(MODEL_NAME) 
    config.save_pretrained('model')

    backbone = TFAutoModel.from_pretrained(MODEL_NAME, config=config)
    backbone.save_pretrained('model')

# NOT MY WORDS

#### EDA of Train set:

1. The train csv has 3911 rows and 8 columns
2. There are 3911 esaays with unique text ids, scored according to six analytic measures: cohesion, syntax, vocabulary, phraseology, grammar, and conventions.
3. Each analytics has 9 scores, range from 1.0 to 5.0 in increments of 0.5.
4. Distribution wise all the six analytics scores show normal distribution.
5. Maximum number of essays are getting a score 3.0 followed by 3.5 as the second mist prevalent score.
6. 1.0 and 5.0 scoring is least prevalent across all analystics

####  Perform Vector Representation, Document Classification
